server/msteams/client_test.go

```diff
@@ -0,0 +1,115 @@
+package msteams
+
+import (
+	"testing"
+	"time"
+
+	"github.com/mattermost/mattermost-plugin-msteams-sync/server/testutils"
+	"github.com/stretchr/testify/assert"
+	msgraph "github.com/yaegashi/msgraph.go/beta"
+)
+
+func TestConvertToMessage(t *testing.T) {
+	teamsUserID := testutils.GetTeamUserID()
+	teamsUserDisplayName := "mockTeamsUserDisplayName"
+	teamsReplyID := "mockTeamsReplyID"
+	content := "mockContent"
+	subject := "mockSubject"
+	reactionType := "mockReactionType"
+	attachmentContent := "mockAttachmentContent"
+	attachmentContentType := "mockAttachmentContentType"
+	attachmentName := "mockAttachmentName"
+	attachmentURL := "mockAttachmentURL"
+	assert := assert.New(t)
+	resp := convertToMessage(&msgraph.ChatMessage{
+		From: &msgraph.IdentitySet{
+			User: &msgraph.Identity{
+				ID:          &teamsUserID,
+				DisplayName: &teamsUserDisplayName,
+			},
+		},
+		ReplyToID: &teamsReplyID,
+		Body: &msgraph.ItemBody{
+			Content: &content,
+		},
+		Subject:              &subject,
+		LastModifiedDateTime: &time.Time{},
+		Attachments: []msgraph.ChatMessageAttachment{
+			{
+				ContentType: &attachmentContentType,
+				Content:     &attachmentContent,
+				Name:        &attachmentName,
+				ContentURL:  &attachmentURL,
+			},
+		},
+		Reactions: []msgraph.ChatMessageReaction{
+			{
+				ReactionType: &reactionType,
+				User: &msgraph.IdentitySet{
+					User: &msgraph.Identity{
+						ID: &teamsUserID,
+					},
+				},
+			},
+		},
+	}, "mockTeamsTeamID", testutils.GetChannelID(), "mockChatID")
+
+	assert.Equal(Message{
+		UserID:          teamsUserID,
+		UserDisplayName: teamsUserDisplayName,
+		ReplyToID:       teamsReplyID,
+		Text:            content,
+		Subject:         subject,
+		LastUpdateAt:    time.Time{},
+		Attachments: []Attachment{
+			{
+				ContentType: attachmentContentType,
+				Content:     attachmentContent,
+				Name:        attachmentName,
+				ContentURL:  attachmentURL,
+			},
+		},
+		Reactions: []Reaction{
+			{
+				Reaction: reactionType,
+				UserID:   teamsUserID,
+			},
+		},
+		ChannelID: testutils.GetChannelID(),
+		TeamID:    "mockTeamsTeamID",
+		ChatID:    "mockChatID",
+	}, *resp)
+}
+
+func TestGetResourceIds(t *testing.T) {
+	for _, test := range []struct {
+		Name           string
+		Resource       string
+		ExpectedResult ActivityIds
+	}{
+		{
+			Name:     "GetResourceIds: With prefix chats(",
+			Resource: "chats(chat123-ID/mockChannel456-ID",
+			ExpectedResult: ActivityIds{
```

Comment:
This expected result is non-sense, the resource doesn't have that shape. If you are testing the happy path the input should be the right one. If you are testing a non-happy path, the error should be captured somehow. Probably the function should be failing somehow with that input.